### PR TITLE
Schemas: List enhancements and support prefect_kind

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Adds the ability to switch between prefect kinds for individual list items. Adds menu items for "Move to bottom" and "Move to top". Abstracts the logic for prefect kind as well as schema property input into compositions for reuse. 

image

### CODE REVIEW

File: src/schemas/components/SchemaFormProperty.vue
Comment (pleek91) [inline src/schemas/components/SchemaFormProperty.vue:1]: Nothing really changed here. Just abstracting pieces of this into compositions for reuse in the list components.

File: src/schemas/components/SchemaFormPropertyArray.vue [hunk 1]
```diff
@@ -58,22 +70,29 @@
...
 }
 
-.schema-form-property-array__handle { @apply
-  cursor-grab
-  mt-2
+.schema-form-property-array__empty { @apply
+  text-subdued
+  text-sm
+  italic;
+  margin-left: 30px;
```
Comment (brandonreid) [inline src/schemas/components/SchemaFormPropertyArray.vue:96]: I'm slightly surprised this worked out to 30px and didn't land on the 8pt grid, but it's an eyeball number regardless.

**Review by brandonreid — APPROVED** [review]

File: src/schemas/compositions/usePrefectKind.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    },
+    set(__prefect_kind) {
+      kindValuesMap[kind.value] = propertyValue.value
+
+      if (__prefect_kind in kindValuesMap) {
+        propertyValue.value = kindValuesMap[__prefect_kind]
+        return
+      }
+
+      propertyValue.value = { __prefect_kind }
```
Comment (znicholasbrown) [inline src/schemas/compositions/usePrefectKind.ts:27]: Making sure I understand this correctly - this is what stores the value for a given kind view when switching between them, right?

File: src/schemas/compositions/useSchemaPropertyInput.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { MaybeRefOrGetter, Ref, computed, toValue } from 'vue'
+import SchemaFormPropertyInput from '@/schemas/components/SchemaFormPropertyInput.vue'
+import SchemaFormPropertyKindJson from '@/schemas/components/SchemaFormPropertyKindJson.vue'
+import { SchemaProperty } from '@/schemas/types/schema'
+import { isPrefectKindValue } from '@/schemas/types/schemaValues'
+import { SchemaValue } from '@/types'
+import { withProps } from '@/utilities/components'
+
+// this is a lot easier to just let typescript infer the return type
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (znicholasbrown) [inline src/schemas/compositions/useSchemaPropertyInput.ts:10]: 👍🏻

**Review by znicholasbrown — APPROVED** [review]

File: src/schemas/components/SchemaFormPropertyArray.vue (hunk 1 above)
Comment (pleek91) [reply src/schemas/components/SchemaFormPropertyArray.vue:96]: This was, special lol. I wanna revisit this at the base component to clean this up. This this keeps me moving for now and it looks fine. Its the width of the drag icon itself plus 8px for gap I believe.

File: src/schemas/compositions/usePrefectKind.ts (hunk 2 above)
Comment (pleek91) [reply src/schemas/compositions/usePrefectKind.ts:27]: Correct, this composition handles caching values when switching and setting the correct value when the user changes the `kind`. Implementation components don't need to care about the value when switching kinds.

**Pull request merged**